content/en/hosting/cht/migration/preparing-for-5.md

```diff
@@ -0,0 +1,153 @@
+
+For instances that use token login,  be sure declare `app_url` in your [settings file]({{< relref "/building/reference/app-settings/token_login/#app_settingsjson-token_login" >}}). This is backwards compatible and safe to do while still on CHT 4.x.
+
+Background information:
+* [Require `app_url` to be set when enabling `token_login`](https://github.com/medic/cht-core/issues/9983)
+
+### Enabling languages via generated docs
+
+{{< callout type="info" >}} Applies to: Instances that enable languages via web interface {{< /callout >}}
+
+As of [CHT 4.2.0]({{< relref "/releases/4_2_0/#technical-improvements" >}}), deployment administrators can enable or disable supported languages [via the settings file]({{< relref "/building/reference/app-settings/#app_settingsjson" >}}). CHT 5.0 removes this web interface to enable or disable languages and requires this to be done via App Settings.
+
+Before upgrading, ensure [your app settings]({{< relref "/building/reference/app-settings/#app_settingsjson" >}}) has the correct languages enabled. This is backwards compatible and safe to do while still on CHT 4.x. If you're unsure if this applies to you, double check your app settings.
+
+If `launguages` is not set, the login screen will show all languages.  Fix it by declaring the `launguages` per above.
+
+Background information:
+* [Deprecate enabling languages through generated docs](https://github.com/medic/cht-core/issues/8157)
+* [Allow app builders to specify disabled languages in config](https://github.com/medic/cht-core/issues/6281)
+
+
+### Declarative Configuration 
+
+{{< callout type="info" >}} Applies to: All Instances {{< /callout >}}
+
+<!-- 
+    todo: add missing TK content on how to make a declarative config
+-->
+tk - all the content ;)  - but upgrade your app settings to be declarative.
+
+Background information:
+* [Make declarative config mandatory](https://github.com/medic/cht-core/issues/5906)
+
+### Increase ecmaVersion ES linting to version 6
+
+{{< callout type="info" >}} Applies to: CHT versions between 4.0 and 4.4  {{< /callout >}}
+
+Deployments running CHT Version 4.0 and 4.4 must upgrade to version 4.5 or later before upgrading to version 5.x.
+
+Background information:
+* [Increase ecmaversion linting for ddocs](https://github.com/medic/cht-core/issues/9202)
+
+### Ensure booleans are used with signoff fields
+
+{{< callout type="info" >}} Applies to: Instances where `needs_signoff` triggers replication of reports  {{< /callout >}}
+
+In some scenarios where `needs_signoff` is set to string `"false"` instead of boolean `false`, replication will happen when it shouldn't.  This is because the string of `"false"` is a truthy value.  
+
+CHT 5.0 now correctly enforces this to be a boolean.  Deployments to make sure that workflows that use `needs_signoff` field have a boolean instead of a string.
```

Comment:
hi @mrjones-plip thanks for documenting this
so technically the field **in the xlsx form** can be a string or a boolean eg needs_signoff -> true or needs_signoff -> "true" basically a boolean or "boolean string" 
the report generated will have a string result either way
not sure if it matters much here though but just wanted to point out that it should work either way

Reply:
Thanks for the tip @jonathanbataire !  I think the text is still accurate as is, but feel free to make a suggestion if you think it needs improvement.  I very much welcome more eyes on this doc! <3